### DISCUSSION TITLE
Periodic community meeting via video?

### DISCUSSION THREAD

**Discussion by djmitche** [GothenburgBitFactory/taskwarrior#3403] [General]
A number of projects I've been involved with have periodic meetings when many of the participants in the project gather to talk about what's going on, discuss new ideas or open questions, and make plans. It's a nice way to generate a bit of social cohesion, too. As an example [Taskcluster has a weekly meeting](https://github.com/taskcluster/taskcluster/discussions/5036) (yes, a large fraction of the open-source projects I work on match `/^Task.*/`).

At this point in the project, I think face-to-face meetings would help us reach a shared understanding of what the biggest needs are, and who can work on them. It's seldom effective to end a GH issue with "maybe someone can work on this?", but if we spend some time talking about a particular piece of work will have clear benefits, sometimes the subsequent awkward silence encourages someone to volunteer.

There are downsides, too:
 * Timezones can make it hard for some people to attend. For example, mid-morning US/Eastern tends to be annoying-but-doable from Europe/Africa through US/Western, but not practical for APAC.
 * Not everyone is able or willing to do real-time video/audio meetings, for a variety of reasons. So making such meetings a component of participating fully in the project can be exclusionary.
 * Meetings aren't very searchable, so decisions made in a meeting can sometimes appear to be "made in secret" from outside.

This came up from a discussion with @wrashi. I think the only other Taskwarrior person I've met face-to-face is @tbabej, a few years ago.

Please, share what you think. Would you participate? What kind of frequency makes sense to you?

**Comment by ryneeverett** [comment]
I'd participate. Maybe biweekly or monthly?

It's hard to know how many people are really watching the discussions to see this -- I don't think anyone is going to chime in to say that they would *not* participate. Maybe it would make sense to ping some people who have actively contributed or been vocal lately just to make sure they're aware that it's happening and they're welcome?

**Reply by wrashi** [reply]
That sounds terrific, @ryneeverett! I've put together a proposal for such a meeting here:
https://github.com/GothenburgBitFactory/taskwarrior/discussions/3403#discussioncomment-9252606

Let me/the community know what you think. 

I think you'd make a terrific community relations officer, if you'd be willing to go through recent discussions and ping people…

**Reply by djmitche** [reply]
Good point regarding people seeing this! I'll ping a few more, in no particular order and with apologies for anyone I've omitted:

 * @dbr
 * @savchenko 
 * @tbabej 
 * @lauft 
 * @felixschurk 
 * @gour
 * @mattsmida
 * @aktaboot

**Reply by wrashi** [reply]
Thanks for bringing these people I was not aware of into the conversation, @djmitche! Your knowledge of project dynamics and community is incredibly useful!

**Comment by wrashi** [comment]
Hey everyone, after the 3.0 upgrade sparked a bunch of discussion the last two weeks, I thought it might be good to get involved in the community, seeing where I could support.
This seems especially appropriate since I've written about about TaskWarrior aimed at non-programmers.
After some discussion with @jtmitche
https://github.com/GothenburgBitFactory/taskwarrior/discussions/3291#discussioncomment-9200290
I went away and thought about the TaskWarrior project and came up with a proposal to help us feel like a community…

It seems to me that
- TaskWarrior is a mature product which works well for the things it does well, so there hasn't been a lot of changes, or need for them.
- Although there are a million to do apps and a half-million project management apps, TaskWarrior fills a unique niche between those two markets.
- With a database back end and better syncing there are new opportunities for extending/improving the functionality of the product.
- There are also new opportunities for users to pick up TW who may not have as much of a technical background.

So, it would benefit all of us (past/present/future users of TV) if there was a time/place to meet, learn how others use TW, ask questions, explore the product road map, hear from devs, etc.

I propose…
- a bi-weekly meeting of 30 minutes to start; we can make it longer later
- 1st meeting should be dominated by the dev team with an update on the 3.0 history, roadmap vision, answer questions from the forum & audience, etc.
- future meetings have 2 parts - user / dev.
- the user section should address
	- "how I use TaskWarrior" - different people giving a 5-minute demo
	- questions from the forum
	- objective: turn this section into video snippets for social media
- the dev section should address
	- roadmap
	- difficulties
	- technical debt
	- help wanted

I can… 
- act as host, including lining up the agenda and keeping the meeting moving 
- personally invite people to present in the meeting
- Get the streaming software going, do a 15-minute pre-meeting to check who's available to participate, audio & video, etc.
- Cut the video into pieces to be posted across social media
- Post meeting notes

I've never done anything like this before, so skills I'll need to learn include
- streaming software like OBS Studio, Stream Yard, or Restream
- setting up an account with Rumble
- creating a 120 second loop clip for before the stream goes live
- creating clips of our discussion for distribution

What does everyone think?
Anyone else willing to help out with this effort?

**Reply by wrashi** [reply]
One element I forgot to add… I'd recommend we try this for 3 months and re-assess instead of a long-term commitment up front.

**Reply by djmitche** [reply]
This sounds great -- getting clips for the socials is a great idea, and not something I have the skills to do.

**Reply by djmitche** [reply]
Also, +1 from me for bi-weekly. That's a nice balance between weekly (which is too often for the velocity of this project) and monthly (which, if someone misses a meeting time, means a loong time between seeing everyone).

**Reply by savchenko** [reply]
> setting up an account with Rumble

Why Rumble? ([Wiki](https://en.wikipedia.org/wiki/Rumble_(company)))

**Reply by wrashi** [reply]
Why not? If you have an objection, I'd love to hear it. Or a better idea, I'm doubly interested.
To me, YouTube feels like Facebook, in the sense that's where my parents hang out. Not the cool kids. Not that I've ever been a cool kid!
I don't have enough knowledge to know if there's a better product out there, which is why I'm so interested in feedback. A number of people have recently told me that Rumble Studio is the only platform product (other than something like OBS) that can stream to multiple platforms, not just its own. That unique capability seems potentially useful.

**Reply by wrashi** [reply]
@djmitche, For the first meeting on Saturday, are you okay with doing most of the talking? Specifically about the work you've done to get to v3.0, what happened, how, and what you see going forward. And then taking questions? My sense is that not all questions need answering in meeting, but at least allowing people to ask and getting back to them in 2 weeks at the next meeting.

**Reply by djmitche** [reply]
Yes, that sounds good.

**Comment by tbabej** [comment]
Happy to join the call. Might not make it every session if it's happening biweekly, but I'll do my best! Suggested agenda by @wrashi looks good, though I suspect keeping it under 30 minutes might be challenging :slightly_smiling_face:

**Reply by djmitche** [reply]
Awesome! I think your historical perspective on the project would be valuable!

**Reply by wrashi** [reply]
Great to have you on board, @tbabej! As for 30 minutes, always leave them wanting more… 😃 What's your timezone?

**Reply by wrashi** [reply]
@tbabej, my memory was jogged this morning of a presentation on TaskWarrior you did I found on YouTube a couple of years ago. It was one of the key prods to get off my butt and write a book. So, thank you for that inspiration! 🎉 

[https://www.youtube.com/watch?v=tijnc65soEI&pp=ygUPdGFza3dhcnJpb3IgdG9t](https://www.youtube.com/watch?v=tijnc65soEI&pp=ygUPdGFza3dhcnJpb3IgdG9t)

**Comment by savchenko** [comment]
Pong @djmitche :-)

Would be happy to join and (if needed) host under company's GMeet account. TZ for Australia can be a little tricky though, could we do a roll call for the timezones?

**Reply by wrashi** [reply]
I'm in the Mountain tz

**Comment by felixschurk** [comment]
Hei, would also be up for joining. :)
My timezone is also Germany, CEST.

**Reply by wrashi** [reply]
Please add your timezone to the roll-call so we can get a sense of what might work best.

**Comment by gour** [comment]
> Please add your timezone to the roll-call so we can get a sense of what might work best.

I'm in Europe, CEST timezone.

**Comment by mattsmida** [comment]
Thank you for the invitation. I'm on the US east coast: timezone is EDT (GMT-4).

**Comment by lauft** [comment]
> Please add your timezone to the roll-call so we can get a sense of what might work best.

Germany, CEST

**Comment by djmitche** [comment]
Looks like
 * @savchenko - UTC+10?
 * @gour - UTC+2
 * @felixschurk - UTC+2
 * @lauft - UTC+2
 * @mattsmida - UTC-4/5
 * @djmitche - UTC-4/5
 * @tbabej - UTC-4/5
 * @wrashi - UTC-6/7
 * @ryneeverett - UTC-4/5

Let's get something scheduled?

**Reply by ryneeverett** [reply]
I'm EST (UTC-4/5) as well.

**Reply by savchenko** [reply]
This should help: https://www.timeanddate.com/worldclock/meetingtime.html?iso=20240501&p1=5&p2=3904&p3=3918&p4=3921

Looks like it will be early morning or late evening for me.

**Reply by wrashi** [reply]
@savchenko, that's very helpful, and exactly the tool I was thinking to use!

Looks like these 2 times are about the best for our group. I'm the furthest west in NA, so I get the early morning. That's good, though, because I'm usually up around 4, anyway. If we aim for a Saturday/Sunday would that work for everyone? Any objections to getting something on the books for **this** weekend?

![CleanShot 2024-05-01 at 05 42 45@2x](https://github.com/GothenburgBitFactory/taskwarrior/assets/489891/38040f7b-7967-4a22-a160-e94cae3b3521)
Use Savchenko's link to see the full day context for this image…

**Reply by djmitche** [reply]
8am is OK for me!

**Reply by lauft** [reply]
> 8am is OK for me!

That would be 14:00 for me. Also OK! 👍🏻 
> ... getting something on the books for **this** weekend?

So 2024-05-04 then?

**Reply by wrashi** [reply]
May the Fourth be with us all! Sounds good to me. Objections anyone?

I'll need to scramble to figure out the technical bits to get it recorded. I'll try to figure out streaming for the next one, which will be the morning after an evening wedding. That'll provide other kinds of challenges. 🥳

**Comment by dbr** [comment]
Sounds like a good idea! I probably wont attend for now as I have very little time to help out here I'm afraid, but, from that perspective:

A short written summary of the meetings would be a really helpful things to have - if written well (which is hard!), they can provide a efficient update on what is going on and the direction of a project - something that can be hard to get any other way.

Some good examples of this would be projects under Academy Software Foundation - they have regular "technical steering committee" meetings, and they store the notes under their respective repo like so:
https://github.com/AcademySoftwareFoundation/openvdb/tree/master/tsc/meetings

The best example I think is Rez:
https://github.com/AcademySoftwareFoundation/rez/tree/main/ASWF/TSC/meeting-notes
very nicely written notes, and has a link to the raw meeting recording (with an automatic text transcript) - this seems like a good balance, probably much less time-consuming than trying to edit down the meeting into clips, and provides most of the same value

**Reply by savchenko** [reply]
@dbr , long time no see! Do you know if Rez notes are auto-generated or typed by someone? If former, I'd be curious about the diarisation pipeline.

**Reply by wrashi** [reply]
@dbr, I don't think we all have to come for the commitment. I, for one, would love to see you came for the comradery. You have "institutional memory" of the project, which makes your contributions valuable.

**Reply by djmitche** [reply]
Notes are definitely a good idea. I was thinking of posting them as a new GitHub discussion topic each time, rather than putting them in a repo, but I'm open as to the approach.

**Comment by wrashi** [comment]
I think the answer in "no," but I'll ask anyway:
##  Does the TaskWarrior project have social media accounts?
Should it?

**Reply by savchenko** [reply]
What's the point at this stage?

**Reply by djmitche** [reply]
I suspect we have a lot of topics for discussion! I certainly have a pile of them. Rather than hash those out here, maybe better to just each keep a list and try to feed the most pertinent/urgent questions into the meetings as time permits (or make new discussions).

**Reply by wrashi** [reply]
@djmitche, I like your focus!

**Comment by wrashi** [comment]
##  First 2024 TaskWarrior Meeting

Our first meeting will be at:
Saturday, May 4, 2024 @ 12:00 UTC
[Using this Google Meet space.](https://meet.google.com/xhg-desb-kap)

Agenda: @djmitche will talk us through v3.0 and take questions. 

We'll aim for 30 minutes to respect everyone's schedules. After event conversation is welcome, though. The space will stay open.

The plan is to record, not stream, this meeting. 

To help coordinate world-wide time zones please look at this [Time & Date World Clock for May 4](https://www.timeanddate.com/worldclock/meetingtime.html?month=5&day=4&year=2024&p1=5&p2=3904&p3=3918&p4=3921&iv=0)

Also, [a calendar event is available](https://calendar.google.com/calendar/event?action=TEMPLATE&tmeid=XzZjc2s0Y2kxNjByM2ViOW04Z3MzaWI5azY0cjMwYmExODkyNDZiYTQ2NHJqZWdxMTg5MjQ4Z2huOG8gd2VzbGV5QG13cGVubmVyLmNvbQ&tmsrc=wesley%40mwpenner.com). Hopefully, I got that set to public properly!

**Reply by wrashi** [reply]
The meeting space is now open for anyone who wants to join early.

**Reply by wrashi** [reply]
Thank you everyone for a terrific first meeting! I really appreciated meeting all of you and loved the way you just jumped in and contributed to the conversation! 

I took a video of most of it and will find a place to host that shortly and link to it here.
I'll also come up with a discussion summary to add to the discussion, as discussed elsewhere in this thread.

As for the next meeting, in about a week I will start a discussion here to collect ideas for the agenda, communicate the time and post the link to the meeting space.

**Reply by wrashi** [reply]
Well, that took longer than I expected, as yesterday developed differently from my plan…

Good news: the video is now edited, uploaded, and available here
https://youtu.be/gcZ6MZYA_88

Bad news: My first try at this didn't go completely smoothly. I didn't capture the whole meeting and I didn't capture my own mic! Lessons to learn…

I'll work on a summary and a clip or two over the next couple of days.

**Reply by djmitche** [reply]
Thanks for organizing that -- it was fantastic to meet so many of you all.
